### PR TITLE
docs: add example to getenv and setenv

### LINKED ISSUE #12311: DOC: clarify `getenv`
So I was actually really confused by the documentation on `getenv` in *expr-env*.

The documentation doesn't specify that you can't prefix the var with a dollar sign.
I.E.

```vim

echo getenv('$VIMRUNTIME')   " -> v:null
echo getenv($VIMRUNTIME)     " -> v:null
echo getenv(VIMRUNTIME)       "  -> raises
echo getenv('VIMRUNTIME')     " expected value

```

In the wild, you'll frequently notice people doing stuff like `expand('$VAR')`,
so having getenv not following those conventional shell rules was confusing
In addition, the documentation on getenv is immediately followed by a section
on `expand` where the example used uses the `$` prefix.

*See below*

I'd be happy to fix this up myself since this is purely documentation but I figured i'd create the
issue before anything

```

environment variable					*expr-env*
--------------------
$VAR			environment variable

The String value of any environment variable.  When it is not defined, the
result is an empty string.

The functions `getenv()` and `setenv()` can also be used and work for
environment variables with non-alphanumeric names.
The function `environ()` can be used to get a Dict with all environment
variables.


						*expr-env-expand*
Note that there is a difference between using $VAR directly and using
expand("$VAR").  Using it directly will only expand environment variables that
are known inside the current Vim session.  Using expand() will first try using
the environment variables known inside the current Vim session.  If that
fails, a shell will be used to expand the variable.  This can be slow, but it
does expand all variables that the shell knows about.  Example: >
	:echo $shell
	:echo expand("$shell")
The first one probably doesn't echo anything, the second echoes the $shell
variable (if your shell supports it).

```

### PR DESCRIPTION
**Pull request by muniter**
fixes #12311
Changes where sent to vim https://github.com/vim/vim/pull/8731

### CODE REVIEW

**Comment by vigoux** [comment]
There must be a vim patch that should be ported to nvim then, could you please port it instead of making the changes directly ?

**Comment by seandewar** [comment]
This'll likely appear in Vim's next runtime files update, whenever that is. 👍

File: runtime/doc/eval.txt [hunk 1]
```diff
@@ -4545,6 +4545,7 @@ getenv({name})						*getenv()*
 		When the variable does not exist |v:null| is returned.  That
 		is different from a variable set to an empty string.
 		See also |expr-env|.
+			getenv('HOME')
```
Comment (seandewar) [inline runtime/doc/eval.txt:4548]: These should be fenced, like:

```suggestion
		See also |expr-env|. Example: <
			getenv('HOME')
```

Bram may have fixed these for you in the merge though. :+1:

**Comment by muniter** [comment]
Will track upstream from when the examples are merged :)

**Comment by seandewar** [comment]
Runtime update is live: https://github.com/vim/vim/commit/6aa57295cfbe8f21c15f0671e45fd53cf990d404

**Comment by justinmk** [comment]
https://github.com/neovim/neovim/wiki/Merging-patches-from-upstream-Vim

**Comment by clason** [comment]
Meaning, the commit message (and PR description) should follow the rules -- look at recent runtime file ports. You can of course amend your commits and reopen this PR instead of creating a new one if you want.

**Pull request closed without merging**